src/main/java/com/example/ludogorieSoft/village/auth/AuthenticationService.java

```diff
@@ -48,14 +93,67 @@ public AuthenticationResponce authenticate(AuthenticationRequest request) {
                             request.getPassword()
                     )
             );
-            var user = administratorRepository.findByUsername(request.getUsername());
+            var user = alternativeUserRepository.findByUsername(request.getUsername());
+            if (!user.isEnabled()) throw new UsernamePasswordException("User not verified!!!");
             var jwtToken = jwtService.generateToken(user);
 
             return AuthenticationResponce.builder()
                     .token(jwtToken)
                     .build();
+        } catch (DisabledException e) {
+            throw new DisabledException("User not verified!!!");
         } catch (Exception e) {
             throw new UsernamePasswordException("Wrong username or password");
         }
     }
+
+    public String verifyVerificationToken(VerificationRequest request) {
+        Optional<VerificationToken> optionalVerificationToken = verificationTokenRepository.findByToken(request.getToken());
+        if (optionalVerificationToken.isEmpty()) throw new ApiRequestException("Invalid token!");
+        VerificationToken verificationToken = optionalVerificationToken.get();
+        AlternativeUser user = verificationToken.getAlternativeUser();
+        if (user.isEnabled()) throw new ApiRequestException("Account already verified!");
+        if (verificationTokenService.isTokenExpired(verificationToken))
+            throw new TokenExpiredException("Expired token!");
+        if (!user.getEmail().equals(request.getEmail()))
+            throw new ApiRequestException("Invalid token!");
+        user.setEnabled(true);
+        alternativeUserRepository.save(user);
+        verificationTokenRepository.delete(verificationToken);
+        return "Your account is verified!";
+    }
+
+    private void checkRegistrationValidations(RegisterRequest request) {
+        if (request.getFullName().isBlank()) throw new ApiRequestException("Full name is required!");
+        if (request.getFullName().length() < 2)
+            throw new ApiRequestException("Full name must be more than 2 characters!");
```

Comment:
The message should say "Full name must be at least 2 characters!" since 2 characters are also allowed. Remove "than" from the @Length message for the fullName in AlternativeUser.